src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/Eth62ProtocolHandler.cs

```diff
@@ -299,6 +299,43 @@ private void Handle(NewBlockMessage newBlockMessage)
             }
         }
 
+        public override void NotifyOfNewBlock(Block block, SendBlockPriority priority)
+        {
+            if (priority == SendBlockPriority.High)
+            {
+                SendNewBlock(block);
+            }
+            else
+            {
+                HintNewBlock(block.Hash, block.Number);
+            }
+        }
+
+        public void SendNewBlock(Block block)
+        {
+            if (Logger.IsTrace) Logger.Trace($"OUT {Counter:D5} NewBlock to {Node:c}");
+            if (block.TotalDifficulty == null)
+            {
+                throw new InvalidOperationException($"Trying to send a block {block.Hash} with null total difficulty");
+            }
+
+            NewBlockMessage msg = new NewBlockMessage();
+            msg.Block = block;
+            msg.TotalDifficulty = block.TotalDifficulty ?? 0;
```

Comment:
block with unknown total difficulty should not be broadcast

Reply:
Good find, I hadn't noticed that. Kind of weird that was there in the first place, since it was checking that it was not null right before there.

Reply:
Looks like it was there to prevent the error about `UInt256` not being able to implicitly cast to `UInt256?`. I changed it to use `TotalDifficulty.Value`, which seems much more clear to me.